ecosystem/nft-metadata-crawler/nft-metadata-crawler-parser/src/utils/uri_parser.rs

```diff
@@ -1,13 +1,51 @@
 // Copyright © Aptos Foundation
 
+use regex::Regex;
+use std::env;
+use url::Url;
+
 pub struct URIParser;
 
 impl URIParser {
     /**
      * Attempts to parse IPFS URI to use dedicated gateway.
      * Returns the original URI if parsing fails.
      */
-    pub fn parse(_uri: String) -> String {
-        todo!();
+    pub fn parse(uri: String) -> String {
+        match Self::parse_uri(uri.clone()) {
+            Ok(parsed_uri) => parsed_uri,
+            Err(_) => uri,
+        }
+    }
```

Comment:
I don't think that you should return the original uri if it fails. Why not just return the error and handle it explicitly somewhere else.

Reply:
yeah, this is a util function. It should not swallow errors